### PR TITLE
shattered-pixel-dungeon: include XDG desktop and icon files

### PR DESCRIPTION
**Pull request by accelbread**
###### Description of changes

Installs the included icons for the package, and adds a .desktop file so it can be launched from a desktop environment's program launcher.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/shattered-pixel-dungeon/default.nix [hunk 1]
```diff
@@ -52,6 +53,20 @@ let
     outputHash = "sha256-i4k5tdo07E1NJwywroaGvRjZ+/xrDp6ra+GTYwTB7uk=";
   };
 
+  desktopFile = writeText "shattered-pixel-dungeon.desktop" ''
```
Comment (fgaz) [inline pkgs/games/shattered-pixel-dungeon/default.nix:56]: Use `makeDesktopItem`

File: pkgs/games/shattered-pixel-dungeon/default.nix [hunk 2]
```diff
@@ -72,6 +87,14 @@ in stdenv.mkDerivation rec {
     makeWrapper ${jre}/bin/java $out/bin/shattered-pixel-dungeon \
       --prefix LD_LIBRARY_PATH : ${libpulseaudio}/lib \
       --add-flags "-jar $out/share/shattered-pixel-dungeon.jar"
+
+    mkdir $out/share/applications
+    cp ${desktopFile} $out/share/applications/shattered-pixel-dungeon.desktop
```
Comment (fgaz) [inline pkgs/games/shattered-pixel-dungeon/default.nix:92]: use `copyDesktopItems`

File: pkgs/games/shattered-pixel-dungeon/default.nix [hunk 3]
```diff
@@ -72,6 +87,14 @@ in stdenv.mkDerivation rec {
...
+    for s in 16 32 48 64 128 256; do
+      mkdir -p $out/share/icons/hicolor/''${s}x$s/apps
+      cp desktop/src/main/assets/icons/icon_$s.png \
+        $out/share/icons/hicolor/''${s}x$s/apps/shattered-pixel-dungeon.png
```
Comment (fgaz) [inline pkgs/games/shattered-pixel-dungeon/default.nix:96]: remove `mkdir`, use `install -Dm644` instead of `cp`

**Review by fgaz — CHANGES_REQUESTED** [review]
Nice!

There are a few things to fix

**Comment by accelbread** [comment]
Thanks! I've updated the PR accordingly.

**Review by fgaz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**